### PR TITLE
Stage2 : refactor buffer overflow

### PR DESCRIPTION
**Pull request by bjjwwang**

### CODE REVIEW

File: svf-llvm/tools/AE/ae.cpp [hunk 1]
```diff
@@ -52,6 +51,9 @@ static Option<bool> AETEST(
     false
 );
 
+static Option<bool> OVERFLOW2(
```
Comment (yuleisui) [inline svf-llvm/tools/AE/ae.cpp:54]: move this to Options.h/cpp

File: svf/include/AE/Core/AbstractState.h [hunk 2]
```diff
@@ -281,13 +283,14 @@ class AbstractState
...
-        return (s32_t) e.lb().getNumeral();
-    }
+    /**
+    * if this NodeID in SVFIR is a pointer, get the pointee type
+    * e.g  arr = (int*) malloc(10*sizeof(int))
+    *      getPointeeType(arr) -> return int
+    * we can set arr[0]='c', arr[1]='c', arr[2]='\0'
+    * @param call callnode of memset like api
+     */
+    const SVFType* getPointeeElement(NodeID id);
```
Comment (yuleisui) [inline svf/include/AE/Core/AbstractState.h:293]: Do we need this? Could we get the type at the use sites?

File: svf-llvm/tools/AE/ae.cpp (hunk 1 above)
Comment (bjjwwang) [reply svf-llvm/tools/AE/ae.cpp]: sorry. I remove it

File: svf/include/AE/Core/AbstractState.h (hunk 2 above)
Comment (bjjwwang) [reply svf/include/AE/Core/AbstractState.h:293]: yes, we need it. Maybe I can figure out a better way to get pointee type.

**Pull request merged**